aggregate-mc.py

```diff
@@ -0,0 +1,128 @@
+from arewefluentyet import data
+import os
+import json
+import argparse
+from datetime import date, timedelta
+
+from helper import read_progress_data, parse_date, read_last_date, \
+    pick_next_revision, get_current_revision, get_revision_date, \
+    switch_to_revision, write_data, is_file_writable, is_dir_readable
+
+DATA_PATH = "./data"
+START_DATE = date(2017, 11, 1)
+FREQUENCY = timedelta(days=7)
+
+
+def extract_progress(dataset):
+    entries = []
+    progress = {}
+
+    for subset in dataset:
+        for path, count in subset.items():
+            entries.append({"file": path, "count": count})
+            ext = os.path.splitext(path)[1]
+            if not ext:
```

Comment:
I don't understand. I'm splitting a path and if the path doesn't contain an extension (for example the file is `foo` instead of `foo.txt`), I wanted to gracefully skip such a file. Where do you want me to move that?

Reply:
before the `entries.append`, as that's unconditional.